### PR TITLE
Write h5py

### PR DESCRIPTION
**Pull request by maxnoe**
This adds methods to write h5py style hdf5 files from pandas dataframes

### CODE REVIEW

**Review by kbruegge — APPROVED** [review]
Maybe not part of this PR. But I think we should make the `read_data` method also be able to read csv files using `pd.read_csv()`. Its quite convenient and doesn't add much  code.

File: fact/io.py [hunk 1]
```diff
@@ -24,12 +26,15 @@
 native_byteorder = native_byteorder = {'little': '<', 'big': '>'}[sys.byteorder]
 
 
-def write_data(df, file_path, key='table'):
+def write_data(df, file_path, key='table', hdf_format='pandas'):
```
Comment (jebuss) [inline fact/io.py]: doesn't a bool use_pandas make more sense?

Comment (jebuss) [reply fact/io.py]: can we you 'data' instead of 'table' as default key? I think its a little less general.

**Review by jebuss — CHANGES_REQUESTED** [review]

File: fact/io.py (hunk 1 above)
Comment (maxnoe) [reply fact/io.py]: I don't really care about the default, data is fine with me.

**Review by jebuss — APPROVED** [review]

**Comment by maxnoe** [comment]
@mackaiver done

**Comment by dneise** [comment]
I am missing some motivational information, as in context. 
What is this feature used for? 
Does this solve an issue?

I mean, technically this is all fine. It does not change API and I believe it is implemented well enough, as it has been reviewed. Also it is of use for a larger group of fact members, so putting it here for everybody to benefit from it also makes sense. 

Maybe a section in the README presenting this new feature to people *outside* TUDO would be good. Alternatively an issue, presenting the problem this feature is going to solve also serves as docu. Or simply a little bit of text in the comment section of this PR not about what ... but about the why would be good.

**Pull request merged**